deepchecks/utils/distribution/drift.py

```diff
@@ -69,27 +70,39 @@ def earth_movers_distance(dist1: Union[np.ndarray, pd.Series], dist2: Union[np.n
 
     Parameters
     ----------
-    dist1 : Union[np.ndarray, pd.Series]
+    dist1: Union[np.ndarray, pd.Series]
         array of numberical values.
-    dist2 : Union[np.ndarray, pd.Series]
+    dist2: Union[np.ndarray, pd.Series]
         array of numberical values to compare dist1 to.
+    margin_quantile_filter: float
+        float in range [0,0.5), representing which margins (high and low quantiles) of the distribution will be filtered
+        out of the EMD calculation. This is done in order for extreme values not to affect the calculation
+        disproportionally. This filter is applied to both distributions, in both margins.
     Returns
     -------
     Any
         the Wasserstein distance between the two distributions.
 
     """
-    unique1 = np.unique(dist1)
-    unique2 = np.unique(dist2)
+    if not isinstance(margin_quantile_filter, Number) or margin_quantile_filter < 0 or margin_quantile_filter >= 0.5:
+        raise DeepchecksValueError(
+            f'margin_quantile_filter expected a value in range [0, 0.5), instead got {margin_quantile_filter}')
 
-    sample_space = list(set(unique1).union(set(unique2)))
+    if margin_quantile_filter != 0:
+        dist1_qt_min = np.quantile(dist1, margin_quantile_filter)
```

Comment:
Think you can make it two lines by `np.quantile(dist1, [margin_quantile_filter, 1-margin_quantile_filter])`. Would probably be more efficient.

Reply:
Thanks!